src/hooks/use-contract-activity.ts

```diff
@@ -51,31 +68,54 @@ export const useContractActivity = (tokenSlug?: string): UseActivityInterface =>
   };
 
   const handleUpdate = async () => {
-    if (activities.length > 0 && !isAllLoaded) {
-      const lastActivityGroup = activities[activities.length - 1].sort((a, b) => b.id - a.id);
+    let newActivities: Array<ActivityGroup> = [];
+    const wasLoading = isLoading;
+    try {
+      if (activities.length > 0 && !isAllLoaded) {
+        const lastActivityGroup = activities[activities.length - 1].sort((a, b) => b.id - a.id);
 
-      if (lastActivityGroup.length > 0) {
-        const lastItem = lastActivityGroup[lastActivityGroup.length - 1];
+        if (lastActivityGroup.length > 0) {
+          const lastItem = lastActivityGroup[lastActivityGroup.length - 1];
 
-        if (lastItem.hash !== lastActivityRef.current) {
-          lastActivityRef.current = lastItem.hash;
+          if (lastItem.hash !== lastActivityRef.current) {
+            lastActivityRef.current = lastItem.hash;
 
-          if (isDefined(lastItem)) {
-            const newActivity = await loadActivity(selectedRpcUrl, selectedAccount, tokenSlug, lastItem);
+            if (isDefined(lastItem) && !isLoading) {
+              setState(prev => ({ ...prev, isLoading: true }));
 
-            if (newActivity.length === 0) {
-              setIsAllLoaded(true);
+              newActivities = await loadActivity(selectedRpcUrl, selectedAccount, tokenSlug, lastItem);
             }
-            setActivities(prev => [...prev, ...newActivity]);
           }
         }
       }
+    } catch (error) {
+      console.error(error);
+    } finally {
+      if (wasLoading) {
+        return;
+      }
+
+      const isAllLoaded = newActivities.length === 0 && !wasLoading;
```

Comment:
so the new activities chunk will be loaded but not set, mb there is no need to load it?

Reply:
ok, sorry, i didnt notice this line